common/hyperscaler/rules/rule.go

```diff
@@ -0,0 +1,225 @@
+package rules
+
+import (
+	"fmt"
+	"sort"
+)
+
+type Labels struct {
+	Labels []string
+}
+
+type Rule struct {
+	Plan                           string
+	PlatformRegion                 string
+	HyperscalerRegion              string
+	EuAccess                       bool
+	Shared                         bool
+	Labels                         map[string]string
+	ContainsInputAttributes        bool
+	ContainsOutputAttributes       bool
+	hyperscalerNameMappingFunction func(string) string
+}
+
+func NewRule() *Rule {
+	return &Rule{
+		hyperscalerNameMappingFunction: getHyperscalerName,
+		Labels:                         make(map[string]string),
+	}
+}
+
+type MatchableAttributes struct {
+	Plan              string `json:"plan"`
+	PlatformRegion    string `json:"platformRegion"`
+	HyperscalerRegion string `json:"hyperscalerRegion"`
+}
+
+func (r *Rule) CalculateLabels() map[string]string {
+	return r.CalculateLabelsWith(getHyperscalerName(r.Plan))
+}
+
+func (r *Rule) CalculateLabelsWith(hyperscalerName string) map[string]string {
+	for _, attr := range AllAttributes {
+		if attr.Getter(r) != "" {
+			r.Labels = attr.ApplyLabel(r, r.Labels)
+		}
+	}
+
+	return r.Labels
+}
+
+func getHyperscalerName(plan string) (result string) {
+	if plan == "aws" || plan == "gcp" || plan == "azure" || plan == "azure_lite" {
+		return plan
+	} else if plan == "trial" {
+		return "aws"
+	} else if plan == "free" {
+		return "aws/azure"
+	} else if plan == "sap-converged-cloud" {
+		return "openstack"
+	} else if plan == "preview" {
+		return "aws"
+	} else {
+		return ""
+	}
+}
+
+func (r *Rule) Matched(attributes *MatchableAttributes) bool {
+
+	if r.Plan != attributes.Plan {
+		return false
+	}
+
+	matched := true
+	for _, attr := range InputAttributes {
+		value := attr.Getter(r)
+		matchableValue := attr.MatchableGetter(attributes)
+		matched = matched && (value == matchableValue || value == ASTERISK || value == "")
+	}
+
+	return matched
+}
+
+func (r *Rule) SetAttributeValue(attribute, value string) (*Rule, error) {
+	for _, attr := range AllAttributes {
+		if attr.Name == attribute {
+			return attr.Setter(r, value)
+		}
+	}
+
+	return nil, fmt.Errorf("unknown attribute %s", attribute)
+}
+
+func (r *Rule) NumberOfInputAttributes() int {
```

Comment:
This is number of non-empty/used/specified InputAttributes.

Reply:
https://github.com/ralikio/kyma-environment-broker/commit/ebb2aa5d7ea7b1a4681344e14419e1664a8d22d0